src/main/java/com/avispl/symphony/dal/avdevices/camera/panasonic/awue150/CameraPanasonicAWUE150Communicator.java

```diff
@@ -0,0 +1,1779 @@
+				responseValueInHex = performCameraControl(command, controllableProperty, value, DeviceURL.CAMERA_PTZ_CONTROL, true);
+				cachedLiveCameraInfo.setFocusPosition(responseValueInHex.substring(Command.FOCUS.length()));
+				try {
+					Thread.sleep(2000);
+				} catch (InterruptedException e) {
+					e.printStackTrace();
+				}
+				retrieveSimultaneous();
+				populateFocusControls(stats, advancedControllableProperties);
+				break;
+			case FOCUS_CONTROL_NEAR:
+				lockFocusControl(controllableProperty);
+				currentUIValue = convertFromApiValueToUIValue(cachedLiveCameraInfo.getFocusPosition(), DeviceConstant.MAX_FOCUS_API_VALUE, DeviceConstant.MIN_FOCUS_API_VALUE,
+						DeviceConstant.FOCUS_UI_API_CONVERT_FACTOR, DeviceConstant.MIN_FOCUS_UI_VALUE) - cachedFocusControlSpeed;
+				if (currentUIValue > DeviceConstant.MAX_FOCUS_UI_VALUE) {
+					currentUIValue = DeviceConstant.MAX_FOCUS_UI_VALUE;
+				}
+				if (currentUIValue < DeviceConstant.MIN_FOCUS_UI_VALUE) {
+					currentUIValue = DeviceConstant.MIN_FOCUS_UI_VALUE;
+				}
+				currentValueInHex = convertFromUIValueToAPIValue(currentUIValue, DeviceConstant.MAX_FOCUS_API_VALUE, DeviceConstant.MIN_FOCUS_API_VALUE, DeviceConstant.MAX_FOCUS_UI_VALUE,
+						DeviceConstant.MIN_FOCUS_UI_VALUE);
+
+				command = Command.HASH.concat(Command.FOCUS).concat(currentValueInHex.toUpperCase());
+				responseValueInHex = performCameraControl(command, controllableProperty, value, DeviceURL.CAMERA_PTZ_CONTROL, true);
+				cachedLiveCameraInfo.setFocusPosition(responseValueInHex.substring(Command.FOCUS.length()));
+				try {
+					Thread.sleep(2000);
+				} catch (InterruptedException e) {
+					e.printStackTrace();
+				}
+				retrieveSimultaneous();
+				populateFocusControls(stats, advancedControllableProperties);
+				break;
+			case FOCUS_CONTROL_SPEED:
+				cachedFocusControlSpeed = (float) Math.ceil(Float.parseFloat(value));
+				populateFocusControls(stats, advancedControllableProperties);
+				break;
+			case AUTO_FOCUS:
+				AutoFocus autoFocus = AutoFocus.getByCode(value);
+				String response = performCameraControl(autoFocus.getApiNameSecond(), controllableProperty, value, DeviceURL.CAMERA_CONTROL, true);
+				autoFocus = AutoFocus.getByApiName2(response);
+				cachedLiveCameraInfo.setAutoFocus(autoFocus);
+				populateFocusControls(stats, advancedControllableProperties);
+				break;
+			default:
+				throw new IllegalStateException(String.format("focus control %s is not supported.", controllableProperty));
+		}
+	}
+
+	/**
+	 * This method is used to lock focus control
+	 *
+	 * @throws IllegalStateException when autofocus is enabled
+	 */
+	private void lockFocusControl(String controllableProperty) {
+		if (AutoFocus.AUTO.equals(cachedLiveCameraInfo.getAutoFocus())) {
+			throw new IllegalStateException(String.format("Failed to control %s, please changing focus mode to manual to control %s", controllableProperty, controllableProperty));
```

Comment:
@param is missing in javadoc, 
it seems that the error text has typo in it -> "changing" instead of "change"

Reply:
I fixed it